internal/connection_manager.go

```diff
@@ -29,30 +28,33 @@ const (
 	serializationVersionMismatch
 )
 
-type connectionManager struct {
-	client              *HazelcastClient
-	connections         map[string]*Connection
-	lock                sync.RWMutex
-	nextConnID          int64
-	connectionListeners atomic.Value
-	listenerLock        sync.Mutex
-}
+type connectionManager interface {
```

Comment:
Using an interface in the same package seems a little bit odd to me. I feel like it would be better if we actually had separate packages for these and the communication was via interfaces.

Reply:
This suits my purpose perfectly. I don't need another package for this. Why introduce one ?

All other services access connection manager over its interface. connectionManagerImpl is hidden to anyone else.

Reply:
I think we should have a `connection` package. Then, this interface will be more meaningful.